Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -997,6 +1027,36 @@ public void run() {
     return true;
   }
 
+  private void updateInformationFromServer(DocumentEvent e) {
+    final Document document = e.getDocument();
+    final VirtualFile file = FileDocumentManager.getInstance().getFile(document);
+    if (file == null) {
+      return;
+    }
+    final String filePath = file.getPath();
```

Comment:
Targets in this file that correspond to regions in other files also become outdated. May be should be handled as well.